client/src/tutorials/BattleCoins/files/lib.rs

```diff
@@ -0,0 +1,252 @@
+
+        // CPI Context
+        let cpi_ctx = CpiContext::new_with_signer(
+            ctx.accounts.token_program.to_account_info(),
+            MintTo {
+                mint: ctx.accounts.reward_token_mint.to_account_info(),
+                to: ctx.accounts.player_token_account.to_account_info(),
+                authority: ctx.accounts.reward_token_mint.to_account_info(),
+            },
+            signer,
+        );
+
+        // Mint 1 token, accounting for decimals of mint
+        let amount = (1u64)
+            .checked_mul(10u64.pow(ctx.accounts.reward_token_mint.decimals as u32))
+            .unwrap();
+
+        mint_to(cpi_ctx, amount)?;
+        Ok(())
+    }
+
+    // Burn token to heal player
+    pub fn heal(ctx: Context<Heal>) -> Result<()> {
+        ctx.accounts.player_data.health = MAX_HEALTH;
+
+        // CPI Context
+        let cpi_ctx = CpiContext::new(
+            ctx.accounts.token_program.to_account_info(),
+            Burn {
+                mint: ctx.accounts.reward_token_mint.to_account_info(),
+                from: ctx.accounts.player_token_account.to_account_info(),
+                authority: ctx.accounts.player.to_account_info(),
+            },
+        );
+
+        // Burn 1 token, accounting for decimals of mint
+        let amount = (1u64)
+            .checked_mul(10u64.pow(ctx.accounts.reward_token_mint.decimals as u32))
+            .unwrap();
+
+        burn(cpi_ctx, amount)?;
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+pub struct CreateMint<'info> {
+    #[account(
+        mut,
+        address = ADMIN_PUBKEY
+    )]
+    pub admin: Signer<'info>,
+
+    // The PDA is both the address of the mint account and the mint authority
+    #[account(
+        init,
+        seeds = [b"reward"],
+        bump,
+        payer = admin,
+        mint::decimals = 9,
+        mint::authority = reward_token_mint,
+
+    )]
+    pub reward_token_mint: Account<'info, Mint>,
+
+    ///CHECK: Using "address" constraint to validate metadata account address
+    #[account(
+        mut,
+        address=find_metadata_account(&reward_token_mint.key()).0
+    )]
+    pub metadata_account: UncheckedAccount<'info>,
+
+    pub token_program: Program<'info, Token>,
+    pub token_metadata_program: Program<'info, Metadata>,
+    pub system_program: Program<'info, System>,
+    pub rent: Sysvar<'info, Rent>,
+}
+
+#[derive(Accounts)]
+pub struct InitPlayer<'info> {
+    #[account(
+        init,
+        payer = player,
+        space = 8 + 8,
+        seeds = [b"player".as_ref(), player.key().as_ref()],
+        bump,
+    )]
+    pub player_data: Account<'info, PlayerData>,
+    #[account(mut)]
+    pub player: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct KillEnemy<'info> {
+    #[account(mut)]
+    pub player: Signer<'info>,
+
+    #[account(
+        mut,
+        seeds = [b"player".as_ref(), player.key().as_ref()],
```

Comment:
Not super important but `.as_ref` on byte string is unnecessary.

```suggestion
        seeds = [b"player", player.key().as_ref()],
```